homework3/semi-uber/Server/api/handlers/user.handlers/userPatchHandler.js

```diff
@@ -0,0 +1,34 @@
+const config = require("config");
+const log4js = require("log4js");
+const logger = log4js.getLogger();
+const { User } = require("../../Schemas/user.schema");
+const bcrypt = require("bcrypt");
+const saltRounds = Number(config.get("saltRounds"));
+const errorHandler = require("../error.handler");
+
+const hashPassword = (password, saltRounds) => {
+  return bcrypt.hash(password, saltRounds);
+};
+
+async function userPatchHandler(req, res) {
+  const { newPassword, id: userID } = req.body;
+  const hashedPass = await hashPassword(newPassword, saltRounds);
```

Comment:
If something goes wrong with password hashing, will it just return some falsy value, or an error might happen? Please check this scenario and make sure you will not get some unexpected errors here

Reply:
Fixed